pulp_ostree/tests/functional/api/test_modify.py

```diff
@@ -1,248 +1,321 @@
-        self.remote = self.remotes_api.create(body)
-        self.addCleanup(self.remotes_api.delete, self.remote.pulp_href)
-
-        repository_sync_data = RepositorySyncURL(remote=self.remote.pulp_href)
-        response = self.repositories_api.sync(self.repo1.pulp_href, repository_sync_data)
-        repo_version_href = monitor_task(response.task).created_resources[0]
-
-        self.repo_version1 = self.versions_api.read(repo_version_href)
-
-    def test_add_ref_and_commit(self):
-        """Copy one commit and one ref from the existing repository."""
-        created_refs = self.refs_api.list(repository_version_added=self.repo_version1.pulp_href)
-        ref = self.refs_api.read(created_refs.to_dict()["results"][0]["pulp_href"])
-        latest_commit = self.commits_api.read(ref.commit)
-
-        configs = self.configs_api.list(repository_version_added=self.repo_version1.pulp_href)
-        cfg = self.configs_api.read(configs.to_dict()["results"][0]["pulp_href"])
-
-        content_to_modify = [ref.to_dict(), latest_commit.to_dict(), cfg.to_dict()]
-
-        self.repo2 = self.repositories_api.create(gen_repo())
-        self.addCleanup(self.repositories_api.delete, self.repo2.pulp_href)
-        modify_repo(self.cfg, self.repo2.to_dict(), add_units=content_to_modify)
-
-        self.repo2 = self.repositories_api.read(self.repo2.pulp_href)
-        self.assertEqual(self.repo2.latest_version_href, f"{self.repo2.pulp_href}versions/1/")
-
-        distribution_data = OstreeOstreeDistribution(
-            **gen_distribution(repository=self.repo2.pulp_href)
-        )
-        response = self.distributions_api.create(distribution_data)
-        distribution = monitor_task(response.task).created_resources[0]
-        self.addCleanup(self.distributions_api.delete, distribution)
-
-        ostree_repo_path = self.distributions_api.read(distribution).base_url
-
-        remote_name = init_local_repo_with_remote(self.remote.name, ostree_repo_path)
-        self.addCleanup(shutil.rmtree, self.remote.name)
-        validate_repo_integrity(
-            self.remote.name,
-            f"{remote_name}:{ref.name}",
-            commits_to_check={latest_commit.checksum},
-            depth=0,
+@pytest.fixture
+def synced_repo_version(
+    ostree_repositories_api_client,
+    ostree_repositories_versions_api_client,
+    ostree_repository_factory,
+    ostree_remote_factory,
+    monitor_task,
+):
+    """
+    Fixture that syncs a Remote ostree Repository,
+    waits until the sync task completes and returns the
+    created repository version object.
+    """
+    repo = ostree_repository_factory()
+    remote = ostree_remote_factory(url=OSTREE_FIXTURE_URL, policy="immediate")
+    result = ostree_repositories_api_client.sync(repo.pulp_href, {"remote": remote.pulp_href})
+    repo_version_href = monitor_task(result.task).created_resources[0]
+    return ostree_repositories_versions_api_client.read(repo_version_href), remote, repo
```

Comment:
Yes, kind of the same but instead of returning the repo we are now returning also the repo_version and remote.
At first, I thought that reusing it (by adding some conditionals or modifying the return) would compromise the readability/maintainability, but thinking again maybe not ... I'll check it!